### DISCUSSION TITLE
How to use oracledb with sqlalchemy 1.4

### DISCUSSION THREAD

**Discussion by jiaulislam** [sqlalchemy/sqlalchemy#8412] [Usage Questions] [answered]
I saw in a discussion #8147 that using oracledb to test but how can I connect to db using oracledb ? I couldn't find any documentation about using oracledb.

I was using below connection string for engine.

```
SA_DATABASE_URI = f"oracle+oracledb://{settings.db_user}:{settings.db_pass}@{settings.tns}"
```

but found exception:
```
 raise exc.NoSuchModuleError(
sqlalchemy.exc.NoSuchModuleError: Can't load plugin: sqlalchemy.dialects:oracle.oracledb
```

**Comment by CaselIT** [comment, marked as answer]
Hi,

Support for orackedb is currently merged only in v2 that's currently in development.

To use orackedb in 1.4 refer to docs at https://python-oracledb.readthedocs.io/en/latest/user_guide/appendix_c.html#python-frameworks-sql-generators-and-orms

**Reply by jiaulislam** [reply]
Hi @CaselIT , Thanks for your reply.

I have seen that documentation but couldn't understand exactly where should I implement that with sqlAlchemy. It would be much helpful if I get an example. Actually I'm not that professional in development yet.

TIA

**Reply by zzzeek** [reply]
hi @jiaulislam -

the same example is given in this post: https://levelup.gitconnected.com/using-python-oracledb-1-0-with-sqlalchemy-pandas-django-and-flask-5d84e910cb19    where it uses the phrase "Then add this to your top level SQLAlchemy file:".     That is, whereever in your program you are using the create_engine() function call, put that code in the same .py file near the top.
